sway-lib-std/src/vec.sw

```diff
@@ -0,0 +1,146 @@
+library vec;
+
+use ::alloc::{alloc, realloc};
+use ::context::registers::heap_ptr;
+use ::intrinsics::size_of;
+use ::mem::{read, write};
+use ::option::Option;
+
+struct RawVec<T> {
+    cap: u64,
+    ptr: u64,
```

Comment:
This is based on Rust's `RawVec`, where `ptr` and `cap` are in that order. https://doc.rust-lang.org/beta/src/alloc/raw_vec.rs.html#52-54

Reply:
Alright, reverted then.